File: examples/automate-google-search/rest/csharp/AutomateGoogleSearch.cs
// Searches Google and extracts result headings using the Browserless /scrape endpoint.
// Uses System.Net.Http and System.Text.Json from the .NET standard library.
// Note: Google may block or CAPTCHA this request — use BQL for more reliable results.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Linq;

const string Token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/scrape?token={Token}";

var payload = new
{
    url = "https://www.google.com/search?q=Browserless+headless+browser",
    elements = new[] { new { selector = "h3" } },
};

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, endpoint);
request.Content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");

var response = await client.SendAsync(request);
string responseBody = await response.Content.ReadAsStringAsync();

var json = JsonDocument.Parse(responseBody);
var results = json.RootElement.GetProperty("data")[0].GetProperty("results");
var titles = results.EnumerateArray()
    .Select(r => r.GetProperty("text").GetString())
    .ToList();
Console.WriteLine(string.Join(", ", titles));

File: examples/automate-google-search/rest/java/AutomateGoogleSearch.java
// Searches Google and extracts result headings using the Browserless /scrape endpoint.
// java.net.http is included in JDK 11+. Gson is used only for JSON parsing.
// Note: Google may block or CAPTCHA this request — use BQL for more reliable results.
//
// Add to pom.xml:
//   <dependency>
//     <groupId>com.google.code.gson</groupId>
//     <artifactId>gson</artifactId>
//     <version>2.10.1</version>
//   </dependency>
//
// Run: javac AutomateGoogleSearch.java && java AutomateGoogleSearch

import com.google.gson.*;
import java.net.URI;
import java.net.http.*;
import java.util.*;

public class AutomateGoogleSearch {
    private static final String TOKEN = "YOUR_API_TOKEN_HERE";

    public static void main(String[] args) throws Exception {
        String endpoint = "https://production-sfo.browserless.io/scrape?token=" + TOKEN;
        String jsonData = """
            {
                "url": "https://www.google.com/search?q=Browserless+headless+browser",
                "elements": [{ "selector": "h3" }]
            }""";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonData))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        JsonArray results = JsonParser.parseString(response.body())
            .getAsJsonObject()
            .getAsJsonArray("data")
            .get(0).getAsJsonObject()
            .getAsJsonArray("results");

        List<String> titles = new ArrayList<>();
        for (JsonElement r : results) {
            titles.add(r.getAsJsonObject().get("text").getAsString());
        }
        System.out.println(titles);
    }
}
